lib/Doctrine/DBAL/Connection.php

```diff
@@ -195,6 +197,7 @@ public function __construct(
 
         if (isset($params['pdo'])) {
             $this->_conn = $params['pdo'];
+            $this->_conn->setAttribute(PDO::ATTR_STATEMENT_CLASS, [PDODriverStatement::class, []]);
```

Comment:
Maybe we should add some logic of the DriverManager here :
- Check that pdo is a PDO instance
- Trigger a deprecation

If we do so, I'm not sure that deprecation in DriverManager will still be useful. Someone using factory would get 2 deprecation.

Reply:
- Moved the deprecation warning to trigger from Connection and removed it from DriverManager
- Added the PDO check